src/org/alanwilliamson/lang/javascript/convert.java

```diff
@@ -126,6 +130,27 @@ public static Object print( Context cx, Scriptable thisObj, Object[] args, Funct
 		thisSession.forceWrite( String.valueOf(args[0]) );
 		return Context.getUndefinedValue();
 	}
+	
+	
+	/*
+	 * This feels super hacky. Someone with better JavaFu, please show me a better way.
+	 * The output follows writeDump for complex values and print() for simple values
+	 */
+	public static Object dump( Context cx, Scriptable thisObj, Object[] args, Function funObj) throws Exception{
+		cfSession thisSession = (cfSession)cx.getThreadLocal("cfsession");
+		
+		try {
+			cfData dat = convert.jsConvert2cfData( (IdScriptableObject) args[0] );
+			String out = cfDUMP.dumpSession(thisSession, dat, null, "", 9999, false, false );
+			thisSession.forceWrite( out );
+			
+		} catch( Exception e ) {
```

Comment:
Why catch the Exception here?

Reply:
Can't cast primitives/simple values to IdScriptableObject, and I couldn't figure out a decent way to test for simple/complex values  :/

I tried using cfData.isSimpleValue() but couldn't get it working going from Object to cfData if I remember right.

Reply:
Ah ok, so it's a ClassCastException you're expecting in that instance.

Use "if ( args[0] instanceof IdScriptableObject )" instead.

Reply:
Aha, I didn't know that was a thing I could check for, thanks Andy!